app/models/concerns/issues/grouped_export.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+module Issues::GroupedExport
+  extend ActiveSupport::Concern
+
+  CSV_OPTIONS       = { col_sep: ';' }
+  HEADERS_SEPARATOR = '⁻-⁻'
+
+  module ClassMethods
+    def grouped_export
+      csvs_by_script = {}
+      grouped_ids    = ids_grouped_by_script_id
+      scripts        = Script.where(id: grouped_ids.keys).pluck(:id, :name).to_h
+
+      grouped_ids.each do |script_id, issue_ids|
+        acc         = {}
+        script_name = scripts[script_id]
+
+        summarized_find_each_for_ids issue_ids do |issue|
+          issue.grouped_export script_name, acc
+        end
+
+        grouped_csvs = build_csvs_for_groups acc
+        filename     = "#{sanitize_filename script_name}.zip"
+
+        csvs_by_script[filename] = tmp_zip_file_content grouped_csvs
+      end
+
+      hash_to_zip csvs_by_script
+    end
+
+    private
+
+      def summarized_find_each_for_ids ids
+        Issue.where(id: ids).select(:id, :description, :data).find_each(batch_size: 100) do |issue|
+          yield issue
+        end
+      end
+
+      def ids_grouped_by_script_id
+        scripts_with_issues = {}
+        select_columns      = ["#{Issue.table_name}.id", "#{Job.table_name}.script_id"]
+
+        all.joins(run: :job).pluck(*select_columns).each do |issue_id, script_id|
+          scripts_with_issues[script_id] ||= []
+          scripts_with_issues[script_id]  << issue_id
```

Comment:
Este no se si lo alinearía, lo dejaría como:
```suggestion
          scripts_with_issues[script_id] << issue_id
```

Si es la única sugerencia sale el merge, después lo tocamos en master.